packages/trader/src/AppV2/Components/TradeParameters/Duration/day.tsx

```diff
@@ -53,31 +56,76 @@ const DayInput = ({
         duration_min_max,
         trade_types,
         contract_type,
+        duration,
+        tick_data,
+        symbol,
+        barrier_1,
     } = useTraderStore();
     const trade_store = useTraderStore();
+    const { addSnackbar } = useSnackbar();
 
     const new_values = {
         duration_unit: 'd',
-        duration: day || '',
-        expiry_time: null,
+        duration: day || duration,
         expiry_type: 'duration',
         contract_type,
+        basis: 'stake',
+        amount: '5',
+        symbol,
     };
+
     const proposal_req = getProposalRequestObject({
         new_values,
         trade_store,
-        trade_type: Object.keys(trade_types)[0],
+        trade_type: contract_type === 'high_low' ? 'PUT' : Object.keys(trade_types)[0],
     });
 
-    const { data: response } = useDtraderQuery<ProposalResponse>(['proposal', JSON.stringify(day)], proposal_req, {
-        enabled: day !== null,
-    });
+    const { data: response } = useDtraderQuery<ProposalResponse>(
+        ['proposal', JSON.stringify(day)],
+        (() => {
+            const request_payload = {
+                ...proposal_req,
+                ...{ symbol },
+            };
+
+            if (barrier_1) {
+                return {
+                    ...request_payload,
+                    barrier: Number(Math.round(tick_data?.quote as number)),
```

Comment:
is it needed to cast Number(..) after Math.round? 
Math.round should already return number, unless there is some edge case here to support,

Reply:
yeah I think Math.round will be enough . I'll change it , thanks.

Reply:
done